addons/mail/wizard/mail_compose_message.py

```diff
@@ -290,6 +291,23 @@ def _compute_mail_server_id(self):
             elif not composer.mail_server_id:
                 composer.mail_server_id = False
 
+    @api.depends('composition_mode', 'model', 'res_ids', 'template_id')
+    def _compute_scheduled_date(self):
+        """ When changing template, update scheduled_date (rendered in comment
+        or raw in mass mode). When removing template, reset value otherwise mail
+        content may not be complete. """
+        for composer in self:
+            res_ids = composer._parse_res_ids(composer.res_ids)
+            if composer.template_id.scheduled_date and composer.composition_mode == 'comment' and len(res_ids) == 1:
+                value = composer._generate_template_for_composer(
+                    composer.template_id,
+                    res_ids,
+                    ('scheduled_date')
+                )[res_ids[0]]
+                composer.scheduled_date = value
+            elif composer.template_id.scheduled_date:
+                composer.scheduled_date = composer.template_id.scheduled_date
```

Comment:
I would add a else condition and write False to reset the value if the template has no scheduled date